### ISSUE TITLE
Another task is already in progress

### ISSUE THREAD

**Issue by hickeng** [vmware/vic#1896]
This is the first time I've actually seen this but we've known it will occur for a while.

This should be fixed via a retry mechanism in the tasks package - this and extraction of error messages is explicitly what that package was intended for.

The problem here is that scratch.vmdk preparation is still underway when we try to create a volume, causing a failure in the appliance reconfigure.
```
time="2016-08-10T05:39:15Z" level=info msg="Refreshing image cache from datastore." 
time="2016-08-10T05:39:15Z" level=info msg="Creating directory [datastore1] VIC/564d3686-b4d2-1f65-6b6e-5c19852997e9/images" 
time="2016-08-10T05:39:15Z" level=info msg="Checking for inconsistent images on http://VCH-0-9650/storage/images/564d3686-b4d2-1f65-6b6e-5c19852997e9" 
time="2016-08-10T05:39:15Z" level=info msg="Creating directory [datastore1] VIC/564d3686-b4d2-1f65-6b6e-5c19852997e9/images/scratch" 
time="2016-08-10T05:39:16Z" level=info msg="Creating directory [datastore1] VIC/564d3686-b4d2-1f65-6b6e-5c19852997e9/images/scratch/imageMetadata" 
time="2016-08-10T05:39:16Z" level=info msg="Creating image scratch ([datastore1] VIC/564d3686-b4d2-1f65-6b6e-5c19852997e9/images/scratch/scratch.vmdk)" 
time="2016-08-10T05:39:16Z" level=debug msg="[BEGIN] [github.com/vmware/vic/pkg/vsphere/disk.(*Manager).CreateAndAttach] [datastore1] VIC/564d3686-b4d2-1f65-6b6e-5c19852997e9/images/scratch/scratch.vmdk" 
time="2016-08-10T05:39:16Z" level=info msg="Create/attach vmdk [datastore1] VIC/564d3686-b4d2-1f65-6b6e-5c19852997e9/images/scratch/scratch.vmdk from parent " 
time="2016-08-10T05:39:19Z" level=debug msg="Mapping vmdk to pci device [datastore1] VIC/564d3686-b4d2-1f65-6b6e-5c19852997e9/images/scratch/scratch.vmdk" 
time="2016-08-10T05:39:19Z" level=debug msg="[BEGIN] [github.com/vmware/vic/pkg/vsphere/disk.findDisk] VirtualMachine:547" 
time="2016-08-10T05:39:19Z" level=debug msg="Looking for attached disk matching filename [datastore1] VIC/564d3686-b4d2-1f65-6b6e-5c19852997e9/images/scratch/scratch.vmdk" 
time="2016-08-10T05:39:20Z" level=debug msg="backing file name [datastore1] VIC/564d3686-b4d2-1f65-6b6e-5c19852997e9/images/scratch/scratch.vmdk" 
time="2016-08-10T05:39:20Z" level=debug msg="Found candidate disk for [datastore1] VIC/564d3686-b4d2-1f65-6b6e-5c19852997e9/images/scratch/scratch.vmdk at [datastore1] VIC/564d3686-b4d2-1f65-6b6e-5c19852997e9/images/scratch/scratch.vmdk" 
time="2016-08-10T05:39:20Z" level=debug msg="[ END ] [github.com/vmware/vic/pkg/vsphere/disk.findDisk] [86.96071ms] VirtualMachine:547" 
time="2016-08-10T05:39:20Z" level=debug msg="[BEGIN] [github.com/vmware/vic/pkg/vsphere/disk.waitForPath] /dev/disk/by-path/pci-0000:03:00.0-scsi-0:0:0:0" 
time="2016-08-10T05:39:20Z" level=debug msg="Waiting for attached disk to appear in /dev/disk/by-path, or timeout" 
time="2016-08-10T05:39:20Z" level=info msg="Attached disk present at /dev/disk/by-path/pci-0000:03:00.0-scsi-0:0:0:0" 
time="2016-08-10T05:39:20Z" level=debug msg="[ END ] [github.com/vmware/vic/pkg/vsphere/disk.waitForPath] [129.982Âµs] /dev/disk/by-path/pci-0000:03:00.0-scsi-0:0:0:0" 
time="2016-08-10T05:39:20Z" level=debug msg="[ END ] [github.com/vmware/vic/pkg/vsphere/disk.(*Manager).CreateAndAttach] [4.040886272s] [datastore1] VIC/564d3686-b4d2-1f65-6b6e-5c19852997e9/images/scratch/scratch.vmdk" 
time="2016-08-10T05:39:20Z" level=debug msg="[BEGIN] [github.com/vmware/vic/pkg/fs.(*Ext4).Mkfs] /dev/disk/by-path/pci-0000:03:00.0-scsi-0:0:0:0" 
time="2016-08-10T05:39:20Z" level=info msg="Creating ext4 filesystem on device /dev/disk/by-path/pci-0000:03:00.0-scsi-0:0:0:0" 
time="2016-08-10T05:39:20Z" level=debug msg="Filesystem created on device /dev/disk/by-path/pci-0000:03:00.0-scsi-0:0:0:0" 
time="2016-08-10T05:39:20Z" level=debug msg="[ END ] [github.com/vmware/vic/pkg/fs.(*Ext4).Mkfs] [469.033639ms] /dev/disk/by-path/pci-0000:03:00.0-scsi-0:0:0:0" 
time="2016-08-10T05:39:20Z" level=debug msg="[BEGIN] [github.com/vmware/vic/pkg/vsphere/disk.(*Manager).Detach] /dev/disk/by-path/pci-0000:03:00.0-scsi-0:0:0:0" 
time="2016-08-10T05:39:20Z" level=info msg="Detaching disk /dev/disk/by-path/pci-0000:03:00.0-scsi-0:0:0:0" 
time="2016-08-10T05:39:20Z" level=debug msg="[BEGIN] [github.com/vmware/vic/pkg/vsphere/disk.findDisk] VirtualMachine:547" 
time="2016-08-10T05:39:20Z" level=debug msg="Looking for attached disk matching filename [datastore1] VIC/564d3686-b4d2-1f65-6b6e-5c19852997e9/images/scratch/scratch.vmdk" 
time="2016-08-10T05:39:20Z" level=debug msg="backing file name [datastore1] VIC/564d3686-b4d2-1f65-6b6e-5c19852997e9/images/scratch/scratch.vmdk" 
time="2016-08-10T05:39:20Z" level=debug msg="Found candidate disk for [datastore1] VIC/564d3686-b4d2-1f65-6b6e-5c19852997e9/images/scratch/scratch.vmdk at [datastore1] VIC/564d3686-b4d2-1f65-6b6e-5c19852997e9/images/scratch/scratch.vmdk" 
time="2016-08-10T05:39:20Z" level=debug msg="[ END ] [github.com/vmware/vic/pkg/vsphere/disk.findDisk] [9.10649ms] VirtualMachine:547" 
time="2016-08-10T05:39:22Z" level=debug msg="[BEGIN] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*StorageHandlersImpl).CreateVolume] storage_handlers.CreateVolume" 
time="2016-08-10T05:39:22Z" level=info msg="Creating directory [datastore1] test/VIC/volumes/test" 
time="2016-08-10T05:39:22Z" level=debug msg="[BEGIN] [github.com/vmware/vic/pkg/vsphere/disk.(*Manager).CreateAndAttach] [datastore1] test/VIC/volumes/test/test.vmdk" 
time="2016-08-10T05:39:22Z" level=info msg="Create/attach vmdk [datastore1] test/VIC/volumes/test/test.vmdk from parent " 
time="2016-08-10T05:39:22Z" level=error msg="vmdk storage driver failed to attach disk: Another task is already in progress." 
time="2016-08-10T05:39:22Z" level=debug msg="[ END ] [github.com/vmware/vic/pkg/vsphere/disk.(*Manager).CreateAndAttach] [21.720865ms] [datastore1] test/VIC/volumes/test/test.vmdk" 
time="2016-08-10T05:39:22Z" level=error msg="storagehandler: VolumeCreate error: Another task is already in progress." 
time="2016-08-10T05:39:22Z" level=debug msg="[ END ] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*StorageHandlersImpl).CreateVolume] [68.75737ms] storage_handlers.CreateVolume" 
time="2016-08-10T05:39:22Z" level=debug msg="[BEGIN] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*StorageHandlersImpl).VolumesList]" 
time="2016-08-10T05:39:22Z" level=debug msg="volumes fetched from list call : []*storage.Volume{}" 
time="2016-08-10T05:39:22Z" level=debug msg="volumes returned from list call : []*models.VolumeResponse(nil)" 
time="2016-08-10T05:39:22Z" level=debug msg="[ END ] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*StorageHandlersImpl).VolumesList] [43.519Âµs] " 
time="2016-08-10T05:39:22Z" level=debug msg="[ END ] [github.com/vmware/vic/pkg/vsphere/disk.(*Manager).Detach] [2.038640564s] /dev/disk/by-path/pci-0000:03:00.0-scsi-0:0:0:0" 
```

**Comment by fdawg4l** [comment]
Looks the similar to #1725

**Comment by hickeng** [comment]
@fdawg4l same symptom, however in this case the VCH is the author of both of the operations rather than conflicting with a reconfigure operations originated by an unknown author.

**Comment by matthewavery** [comment]
Looks like #2013 is also suffering from this issue.

**Comment by matthewavery** [comment]
solved with pr #2141
